src/crangler/c_wrangler.cpp

```diff
@@ -0,0 +1,474 @@
+/*******************************************************************\
+
+Module: C Wrangler
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+/// \file
+/// C Wrangler
+
+#include "c_wrangler.h"
+
+#include "c_defines.h"
+#include "ctokenit.h"
+#include "mini_c_parser.h"
+
+#include <util/cprover_prefix.h>
+#include <util/exception_utils.h>
+#include <util/file_util.h>
+#include <util/optional.h>
+#include <util/prefix.h>
+#include <util/run.h>
+#include <util/string_utils.h>
+#include <util/suffix.h>
+#include <util/tempdir.h>
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <unordered_map>
+
+struct c_wranglert
+{
+  // sources and preprocessing
+  std::vector<std::string> source_files;
+  std::vector<std::string> includes;
+  std::vector<std::string> defines;
+
+  // transformations
+  struct contract_clauset
+  {
+    std::string clause;
+    std::string content;
+    contract_clauset(std::string _clause, std::string _content)
+      : clause(std::move(_clause)), content(std::move(_content))
+    {
+    }
+  };
+
+  struct loop_invariantt
+  {
+    std::string loop_type;
+    std::string identifier;
+    std::string content;
+    loop_invariantt(
+      std::string _loop_type,
+      std::string _identifier,
+      std::string _content)
+      : loop_type(std::move(_loop_type)),
+        identifier(std::move(_identifier)),
+        content(std::move(_content))
+    {
+    }
+  };
+
+  struct assertiont
+  {
+    std::string identifier;
+    std::string content;
+    assertiont(std::string _identifier, std::string _content)
+      : identifier(std::move(_identifier)), content(std::move(_content))
+    {
+    }
+  };
+
+  struct functiont
+  {
+    // should be variant to preserve ordering
+    std::vector<contract_clauset> contract;
+    std::vector<loop_invariantt> loop_invariants;
+    std::vector<assertiont> assertions;
+    optionalt<std::string> stub;
+    bool remove_static = false;
+  };
+
+  using functionst = std::map<std::string, functiont>;
+  functionst functions;
+
+  // output
+  std::string output;
+
+  void configure_sources(const jsont &);
+  void configure_functions(const jsont &);
+  void configure_output(const jsont &);
+};
+
+void c_wranglert::configure_sources(const jsont &config)
+{
+  // look for the 'sources' key
+  auto sources = config["sources"];
```

Comment:
There aren't a lot of comments in this code, but then this one stating the obvious. Perhaps I'm missing something and it's not actually doing the obvious?